command/agent/agent.go

```diff
@@ -418,6 +418,16 @@ func (a *Agent) setupServer() error {
 				},
 			},
 		}
+		// Resolve ServiceCheck addresses
```

Comment:
Can you pull these into three methods, `getHTTPAddr(preferBind bool)`, `getRPCAddr(preferBind bool)` and `getSerfAddr(preferBind bool)` and use those methods for the service and check. 

https://github.com/hashicorp/nomad/blob/master/command/agent/agent.go#L175-L188

Then add a config option to the `Consul` config `checks_use_advertise bool`. This will default to false, but if people want the checks to use the advertise they can set it to true.

Please add tests and documentation to the website